### ISSUE TITLE
Sharing with multiple users

### ISSUE THREAD

**Issue by aimless55** [google/cloud-print-connector#333]
If I want to share cloud printers with multiple users, can `share_scope` take a comma separated list? ie

    "share_scope": "foo@example.com,bar@example.com"

Or do I need to create a group with these users and share with that group?

Similarly, if I want to share with my entire G Suite domain can I do something like:

    "share_scope": "*@example.com"

Thanks!

**Comment by sicklittlemonkey** [comment]
* Tubleweed * ...

Hi!

I'm not an expert in this area, but I tested some things for you. A group email address is a common way to share to multiple users, so that'll work if you use it.

The connector doesn't handle multiple email addresses or anything specially - it just passes the string to the Google Cloud Print API, which says the scope is the  "Email of the user or group or domain name to share the printer with".

Since you have a "G Suite domain", and the API mentions a domain name, it sounds like you could use just the domain name. Unfortunately I'm not able to test that, but please give it a try.

I tested the examples you give above using the [API tool](https://www.google.com/cloudprint/simulate.html) and they failed.

Cheers,
Nick.

**Comment by pigmonkey** [comment]
I use cloud-print-connector with G Suite. To share a printer with the entire domain, I created an everyone@example.com group which automatically includes all members of the domain and used that as my `share_scope`.

After sharing the printers, I had to add myself as an owner of the group and then visit https://www.google.com/cloudprint#printers, where I was prompted to accept the share on behalf of all group members. I couldn't accept the share before I set myself as a group owner, despite being a super admin on the domain.

**Comment by jay0lee** [comment]
@pigmonkey: yes, that's expected behavior, only group owners can accept a printer shared with group. However, if you add your cloud print user as an owner of the group the group share will be automatic since the user owns both the printer and group.

**Comment by seandarcy** [comment]
I want to set up multiple users, but we're not using G Suite. So I don't believe I can create a "domain".

Can I run multiple instances of gcp on the same box, each with a different share_scope ?  E.g., sean@gmail.com  in one instance and fred@yahoo.com in another instance ?

**Comment by seandarcy** [comment]
A little self help works wonders.

Go to google groups and set up a new group, with some email : myprinters@googleproups.com.
Send invites. Accept invites. In gcp:
  "share_scope": "myprinters@googlegroups.com",

And now you have multiple users !

**Comment by seandarcy** [comment]
Not so fast. Now only one member of the group sees the printers, Three members, 2 are owners, 1 a manager. One of the owners can see and use the printers, but not the other owner or the manager.

All three can access the group page, and are listed as members,

???

**Comment by bblietz** [comment]
Has everyone accepted the invite to the Google Group?

**Comment by seandarcy** [comment]
Yes, they all accepted. Each can access the google groups page.  And each shows as a member on the groups page:

"Showing 3 of 3 members"

**Comment by BorysekOndrej** [comment]
I'm experiencing similar issue as @seandarcy, though I doubt it's problem with the connector itself.
In my case, the only user who can't see the printer is the user, who is in the group and outside of the domain. I guess I will contact the support the and try to find out whats wrong.
